airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/connector_runner.py

```diff
@@ -5,192 +5,128 @@
 
 import json
 import logging
+import os
 from pathlib import Path
-from typing import Iterable, List, Mapping, Optional
+from typing import List, Mapping, Optional
 
-import docker
-from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog, OrchestratorType
-from airbyte_cdk.models import Type as AirbyteMessageType
-from docker.errors import ContainerError, NotFound
-from docker.models.containers import Container
+import anyio
+import dagger
+from airbyte_protocol.models import AirbyteMessage, ConfiguredAirbyteCatalog, OrchestratorType
+from airbyte_protocol.models import Type as AirbyteMessageType
+from connector_acceptance_test.utils import SecretDict
 from pydantic import ValidationError
 
 
 class ConnectorRunner:
     def __init__(
         self,
         image_name: str,
-        volume: Path,
+        base_path: Path,
         connector_configuration_path: Optional[Path] = None,
         custom_environment_variables: Optional[Mapping] = {},
     ):
-        self._client = docker.from_env(timeout=120)
-        try:
-            self._image = self._client.images.get(image_name)
-        except docker.errors.ImageNotFound:
-            print("Pulling docker image", image_name)
-            self._image = self._client.images.pull(image_name)
-            print("Pulling completed")
+        self.base_path = base_path
+        self.image_name = image_name
         self._runs = 0
-        self._volume_base = volume
         self._connector_configuration_path = connector_configuration_path
         self._custom_environment_variables = custom_environment_variables
+        self.dagger_connection = dagger.Connection(dagger.Config())
 
-    @property
```

Comment:
All this code is not useful anymore as the Dagger API makes this straightforward.

Reply:
This is great!